core/src/autogluon/core/hpo/executors.py

```diff
@@ -193,6 +196,10 @@ def register_resources(
             if minimum_model_num_gpus > 0:
                 num_jobs_in_parallel_with_gpu = num_gpus // minimum_model_num_gpus
             num_jobs_in_parallel = min(num_jobs_in_parallel_with_mem, num_jobs_in_parallel_with_cpu, num_jobs_in_parallel_with_gpu)
+            if k_fold is not None and k_fold > 0:
+                num_jobs_in_parallel = min(num_jobs_in_parallel, self.hyperparameter_tune_kwargs.get("num_trials", math.inf) * k_fold)
```

Comment:
nit: Have a variable assigned to `max_models` so it is easier to understand this logic.

Also, this might get more confusing if we add nested bagging or repeated k-fold in the logic.